Phosphor/src/main/java/edu/columbia/cs/psl/phosphor/instrumenter/ConfigurationEmbeddingMV.java

```diff
@@ -0,0 +1,108 @@
+package edu.columbia.cs.psl.phosphor.instrumenter;
+
+import edu.columbia.cs.psl.phosphor.Configuration;
+import edu.columbia.cs.psl.phosphor.struct.harmony.util.Set;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+
+import static org.objectweb.asm.Opcodes.ACC_PUBLIC;
+import static org.objectweb.asm.Opcodes.ACONST_NULL;
+import static org.objectweb.asm.Opcodes.ASM9;
+import static org.objectweb.asm.Opcodes.DUP;
+import static org.objectweb.asm.Opcodes.INVOKEINTERFACE;
+import static org.objectweb.asm.Opcodes.INVOKESPECIAL;
+import static org.objectweb.asm.Opcodes.POP;
+
+public class ConfigurationEmbeddingMV extends MethodVisitor  {
+    public ConfigurationEmbeddingMV(MethodVisitor mv) {
+        super(ASM9, mv);
+    }
+
+    // Embed initialized Configuration class into class file.
+    @Override
+    public void visitFieldInsn(int opcode, String owner, String name, String descriptor) {
+        if (opcode == Opcodes.PUTSTATIC) {
+            Type type = Type.getType(descriptor);
+            Object fieldValue;
+            if (name.equals("IS_JAVA_8")) {
+                fieldValue = false;
+            } else {
+                try {
+                    Field f = Configuration.class.getField(name);
+                    if ((f.getModifiers() & ACC_PUBLIC) == 0) {
+                        super.visitFieldInsn(opcode, owner, name, descriptor);
+                        return;
+                    }
+                    f.setAccessible(true);
+                    fieldValue = f.get(null);
+                    if (fieldValue instanceof Class) {
+                        fieldValue = Type.getType((Class<?>) fieldValue);
+                    }
+                } catch (NoSuchFieldException | IllegalAccessException e) {
+                    super.visitFieldInsn(opcode, owner, name, descriptor);
+                    return;
+                }
+            }
+            switch (type.getSort()) {
+                case Type.LONG:
+                case Type.DOUBLE:
+                    super.visitInsn(POP);
```

Comment:
Should this be POP2?

Reply:
The second operand will be poped at line 60

Reply:
If a two-word operands is on the stack (aka long or double), it must be removed with `POP2`, not `POP`

Reply:
I see. I thought long double can be removed with two `POP`s. I'll fix this soon.